Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -197,8 +286,31 @@ public struct HTTPHeaders: CustomStringConvertible {
     //      recommended.
     /// - Parameter value: The header field value to add for the given name.
     public mutating func add(name: String, value: String) {
-        let keyLower = name.lowercased()
-        storage[keyLower] = (storage[keyLower] ?? [])  + [(name, value)]
+        self.add(headerName: .string(string: name), value: value)
+    }
+
+    /// Add a header name/value pair to the block.
+    ///
+    /// This method is strictly additive: if there are other values for the given header name
+    /// already in the block, this will add a new entry. `add` performs case-insensitive
+    /// comparisons on the header field name.
+    ///
+    /// - Parameter name: The header field name. For maximum compatibility this should be an
+    ///     ASCII string. For future-proofing with HTTP/2 lowercase header names are strongly
+    //      recommended.
+    /// - Parameter value: The header field value to add for the given name.
+    public mutating func add(name: StaticString, value: String) {
```

Comment:
This won't work the way you want: I quickly asked @weissi and he reckons that the compiler won't ever select this method over its other overload. Can you try replacing the body of this method with `fatalError` and running the tests to see if you ever actually execute this code? If you don't, this probably needs a different set of labels to make the compiler actually choose it.